src/js/containers/account/WithLatestFy.jsx

```diff
@@ -53,3 +55,35 @@ WithLatestFy.propTypes = {
 };
 
 export default WithLatestFy;
+
+// TODO: Refactor existing consumers of WithLatestFy to use this custom hook
+export const useLatestAccountData = () => {
+    const [submissionPeriods, setSubmissionPeriods] = useState([]);
+    const { dataAsOf } = useSelector((state) => state.account);
+    const request = useRef();
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (!dataAsOf) {
```

Comment:
We are also setting the latest submissions in redux, we should be checking if that data exists as well to prevent fetching data we already have.